system/View/Parser.php

```diff
@@ -0,0 +1,296 @@
+/**
+ * CodeIgniter
+ *
+ * An open source application development framework for PHP
+ *
+ * This content is released under the MIT License (MIT)
+ *
+ * Copyright (c) 2014 - 2016, British Columbia Institute of Technology
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @package	CodeIgniter
+ * @author	CodeIgniter Dev Team
+ * @copyright	Copyright (c) 2014 - 2016, British Columbia Institute of Technology (http://bcit.ca/)
+ * @license	http://opensource.org/licenses/MIT	MIT License
+ * @link	http://codeigniter.com
+ * @since	Version 3.0.0
+ * @filesource
+ */
+use Config\Services;
+use CodeIgniter\Log\Logger;
+
+/**
+ * Class Parser
+ *
+ *  ClassFormerlyKnownAsTemplateParser 
+ * 
+ * @todo Views\Parser_Test
+ * @tofo Common::parse
+ * @todo user guide
+ * @todo options -> delimiters
+ *
+ * @package CodeIgniter\View
+ */
+class Parser extends View
+{
+
+	/**
+	 * Left delimiter character for pseudo vars
+	 *
+	 * @var string
+	 */
+	public $leftDelimiter = '{';
+
+	/**
+	 * Right delimiter character for pseudo vars
+	 *
+	 * @var string
+	 */
+	public $rightDelimiter = '}';
+
+	// --------------------------------------------------------------------
+
+	/**
+	 * Constructor
+	 *
+	 * @param string $viewPath
+	 * @param type $loader
+	 * @param bool $debug
+	 * @param Logger $logger
+	 */
+	public function __construct(string $viewPath = null, $loader = null, bool $debug = null, Logger $logger = null)
+	{
+		parent::__construct($viewPath, $loader, $debug, $logger);
+	}
+
+	// --------------------------------------------------------------------
+
+	/**
+	 * Parse a template
+	 *
+	 * Parses pseudo-variables contained in the specified template view,
+	 * replacing them with any data that has already been set.
+	 *
+	 * @param string $view
+	 * @param array  $options  
+	 * @param bool $saveData
+	 *
+	 * @return string
+	 */
+	public function render(string $view, array $options = null, bool $saveData = false): string
```

Comment:
Just a thought, but would it make sense to extend the View class itself, or would that get in the way and make things more complex? Just looking at this method looks an awful lot like the View class'.